tests/core/commands/test_renku.py

```diff
@@ -17,11 +17,82 @@
 # limitations under the License.
 """Module tests."""
 
-from __future__ import absolute_import, print_function
+import pytest
+
+from renku.command.command_builder.command import Command
+from renku.core import errors
+from renku.domain_model.project import Project
+from renku.domain_model.provenance.agent import Person
 
 
 def test_version():
     """Test version import."""
     from renku import __version__
 
     assert __version__
+
+
+def test_minimum_version(mocker, tmpdir):
+    """Test minimum version required."""
+
+    def _mock_database_project(project):
+        def mocked_getter(self, key):
+            if key == "project":
+                return project
+            return getattr(self, key)
+
+        return mocked_getter
+
+    with tmpdir.as_cwd():
+        # NOTE: Check doesn't raise with identical version
+        mocker.patch("renku.domain_model.project.Project.minimum_renku_version", "1.0.0")
+        project = Project(creator=Person(name="John Doe", email="jd@example.com"), name="testproject")
+        mocker.patch(
+            "renku.command.command_builder.database_dispatcher.Database.__getitem__", _mock_database_project(project)
+        )
+        mocker.patch("renku.version.__version__", "1.0.0")
+
+        try:
+            Command().command(lambda: "").with_database(write=False).build().execute()
+        except errors.MinimumVersionError:
+            assert False, "Shouldn't raise an exception"
```

Comment:
Maybe use `pytest.fail` to be more explicit. There are more similar cases in this function.